src/System.CommandLine.Tests/Binding/ModelBinderTests.cs

```diff
@@ -79,6 +79,39 @@ public class ModelBinderTests
             valueReceivedValue.Should().Be(expectedValue);
         }
 
+        [Theory]
+        [InlineData(typeof(FileInfo), "MyFile.cs")]
+        public void Command_arguments_are_bound_by_name_to_complex_constructor_parameters(
+           Type type,
+           string commandLine)
+        {
+            var targetType = typeof(ClassWithCtorParameter<>).MakeGenericType(type);
+            var binder = new ModelBinder(targetType);
+
+            var command = new Command("the-command")
+            {
+                new Argument
+                {
+                    Name = "value",
+                    ArgumentType = type
+                }
+            };
+
+            var bindingContext = new BindingContext(command.Parse(commandLine));
+
+            var instance = binder.CreateInstance(bindingContext);
+
+            object valueReceivedValue = ((dynamic)instance).Value;
+            var expectedValue = new FileInfo(commandLine);
+
+            valueReceivedValue.Should().BeOfType<FileInfo>();
+            var fileInfoValue = valueReceivedValue as FileInfo;
+            fileInfoValue.FullName.Should().Be(expectedValue.FullName);
+            // The following fails when it attempts to compare the Length of the file. I have
+            // no idea why this previously worked. 
+            //valueReceivedValue.Should().BeEquivalentTo(expectedValue);
```

Comment:
Doing a property-by-property comparison on a `FileInfo` is probably not useful. I would just compare the properties that matter.

Reply:
I just dropped that line of code. I was hesitant to just throw away a test, but the presence of the file is not relevant to this test.